device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubListener.java

```diff
@@ -60,8 +60,10 @@
      * @param connectionId the Id of the connection, used to identify which of possible many reconnection attempts
      *                     this event belongs to.
      * @param deviceId the Id of the device that the session belongs to
+     * @param shouldReconnect false if this session was lost because the user opted to unregister the client from
```

Comment:
I think another way of saying this is whether the disconnect was desired by the user, or the consequence of an issue where we'd want to retry connection.